modules/plottinggl/glsl/pcp_lines.frag

```diff
@@ -31,13 +31,13 @@ void main() {
             res.rgb = 1 - res.rgb;
         }
         if (filtered == 1) {
-            res.xyz = mix(res.xyz, filterColor.xyz, filterIntensity);
-            res.a = filteredAlpha;
+            res = mix(res, filterColor, filterIntensity);
+            res.a = filterColor.w;
```

Comment:
No need to assign it afterwards (otherwise the mix above with all components won’t make a difference)

Can’t we simply assign a filtered color instead of using mix with a filter intensity?

Reply:
> No need to assign it afterwards (otherwise the mix above with all components won’t make a difference)

True, removed that row!

> Can’t we simply assign a filtered color instead of using mix with a filter intensity?

We sure can but it won't give the same result. The advantage of mixing the two colors is that you can get a hint of the filtered lines original color if you want. Another way could be to just modify the saturation of the filtered lines.

Reply:
>> No need to assign it afterwards (otherwise the mix above with all components won’t make a difference)

> True, removed that row!

Alright, seems we were both wrong. The row is needed or else alpha value of the filtered lines will be:
`res.w = mix(unfilteredAlpha, filteredAlpha, filterIntensity)`
which is not the same as
`res.w = filteredAlpha`

Reply:
That’s why I think we could assign a color. Saturation sounds like a good idea though :)

Reply:
I don't really understand what you mean with the assigned color. Isn't that what were doing right now?

Reply:
Now we are mixing the two colors. I was thinking of simply assigning the color instead:
res = filteredColor;

Reply:
> The advantage of mixing the two colors is that you can get a hint of the filtered lines original color if you want.

I think it's still a nice feature to be able to see the original colors of the filtered lines.